processor/routingprocessor/processor_test.go

```diff
@@ -349,6 +349,55 @@ func TestTraces_RoutingWorks_ResourceAttribute(t *testing.T) {
 	})
 }
 
+func TestTraces_RoutingWorks_ResourceAttribute_DropsRoutingAttribute(t *testing.T) {
+	defaultExp := &mockTracesExporter{}
+	tExp := &mockTracesExporter{}
+
+	host := &mockHost{
+		Host: componenttest.NewNopHost(),
+		GetExportersFunc: func() map[config.DataType]map[config.ComponentID]component.Exporter {
+			return map[config.DataType]map[config.ComponentID]component.Exporter{
+				config.TracesDataType: {
+					config.NewComponentID("otlp"):   defaultExp,
+					config.NewComponentID("otlp/2"): tExp,
+				},
+			}
+		},
+	}
+
+	exp := newProcessor(zap.NewNop(), &Config{
+		AttributeSource:              resourceAttributeSource,
+		FromAttribute:                "X-Tenant",
+		DropRoutingResourceAttribute: true,
+		DefaultExporters:             []string{"otlp"},
+		Table: []RoutingTableItem{
+			{
+				Value:     "acme",
+				Exporters: []string{"otlp/2"},
+			},
+		},
+	})
+	require.NoError(t, exp.Start(context.Background(), host))
+
+	tr := pdata.NewTraces()
+	rm := tr.ResourceSpans().AppendEmpty()
+	rm.Resource().Attributes().InsertString("X-Tenant", "acme")
+	rm.Resource().Attributes().InsertString("attr", "acme")
+
+	assert.NoError(t, exp.ConsumeTraces(context.Background(), tr))
+	assert.Equal(t, 1, tExp.getTraceCount(),
+		"metric should be routed to non default exporter",
+	)
+	require.Len(t, tExp.traces, 1)
+	require.Equal(t, 1, tExp.traces[0].ResourceSpans().Len())
+	attrs := tExp.traces[0].ResourceSpans().At(0).Resource().Attributes()
+	_, ok := attrs.Get("X-Tenant")
+	assert.False(t, ok, "routing attribute should be dropped")
+	v, ok := attrs.Get("attr")
+	assert.True(t, ok, "non routing attributes shouldn't be dropped")
+	assert.Equal(t, "acme", v.StringVal())
```

Comment:
Wouldn't this assertion already cover the assertion that came just before this one?

Reply:
I don't believe so. The other one check for the attribute for existence while this one checks its value. It might catch something tampering with the value in the future.

Reply:
True. Better to have it redundant today than a missed coverage in the future.